web/scripts/generate-llm-files.mjs

```diff
@@ -0,0 +1,483 @@
+ */
+async function processBlogFiles() {
+  let blogSectionContent = `## Blogposts\n`
+  const blogPostFiles = globSync('*.{md,mdx}', {
+    cwd: BLOG_DIR,
+    nodir: true,
+    ignore: ['_*.md', '_*.mdx', 'authors.yml', 'components/**'],
+  })
+
+  const blogPostsData = []
+
+  for (const file of blogPostFiles) {
+    const dateMatch = validateBlogFileDate(file)
+    if (!dateMatch) {
+      continue
+    }
+    const dateString = dateMatch[1]
+    const absoluteFilePath = path.join(BLOG_DIR, file)
+    try {
+      const rawContent = await fs.readFile(absoluteFilePath, 'utf8')
+      const { attributes } = fm(rawContent)
+
+      const title = extractBlogPostTitle(attributes, file)
+      const linkPath = constructBlogUrl(file)
+
+      if (linkPath) {
+        blogPostsData.push({ title, dateString, linkPath })
+      } else {
+        console.warn(
+          `Skipping blog post ${file}, does not match YYYY-MM-DD-name.md(x) naming convention`
+        )
+      }
+    } catch (fileReadError) {
+      // This is intentionally not re-thrown to allow the script to continue
+      // and generate a partial file, even if some source files have errors.
+      console.error(
+        `Error reading or processing blog file ${absoluteFilePath}:`,
+        fileReadError
+      )
+    }
+  }
+
+  blogPostsData.sort((a, b) => b.dateString.localeCompare(a.dateString))
+
+  if (blogPostsData.length > 0) {
+    for (const post of blogPostsData) {
+      blogSectionContent += `- [${post.title}](${post.linkPath})\n`
+    }
+    return blogSectionContent.trim()
+  } else {
+    return ''
+  }
+}
+
+/**
+ * Validates a blog post filename against the 'YYYY-MM-DD-slug' format.
+ * Logs a warning for files that don't match and aren't explicitly ignored.
+ * @param {string} file - The filename to validate.
+ * @returns {RegExpMatchArray | null} The result of the match, or null if it doesn't match.
+ */
+function validateBlogFileDate(file) {
+  const dateMatch = file.match(/^(\d{4}-\d{2}-\d{2})-.*\.(mdx|md)$/)
+  if (dateMatch) {
+    return dateMatch
+  }
+
+  // If it doesn't match, check if it's a file type we should warn about.
+  const isIgnoredType =
+    file === 'authors.yml' ||
+    file.startsWith('_') ||
+    file.includes('components/')
+
+  if (!isIgnoredType) {
+    console.warn(
+      `Skipping file in web/blog, does not match YYYY-MM-DD-name.md(x) naming convention or is an ignored type: ${file}`
+    )
+  }
+  return null
+}
+
+/**
+ * Extracts the title for a blog post.
+ * It first checks for a title in the front-matter attributes. If not found,
+ * it generates a title from the filename.
+ * @param {object} attributes - The front-matter attributes from the blog post file.
+ * @param {string} filename - The filename of the blog post.
+ * @returns {string} The extracted or generated title.
+ */
+function extractBlogPostTitle(attributes, filename) {
+  if (attributes.title) {
```

Comment:
Heads up! This is not the same logic you had before! I think something got lost in the refactor


```suggestion
  if (!attributes.title) {
```

Reply:
Got it!